docs/DeveloperGuide.md

```diff
@@ -8,8 +8,189 @@
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Database
+
+The database reads in SEP module data from data.csv and stores the useful data into an ArrayList of module mappings and universities.
+
+Upon starting easySEP, the DatabaseStorage will load each line from data.csv, parse the line using DatabaseParser, and store the data into the Database.
+
+Relevant exceptions are thrown when there are unexpected scenarios. For instance, if data.csv cannot be found at the given file path, a FileNotFoundException is thrown.
```

Comment:
You may consider putting data.csv in a code block, e.g. `data.csv` so that it helps to enhance the readability / highlight the importance of the file?